python/ray/experimental/channel/cached_channel.py

```diff
@@ -107,3 +107,9 @@ def close(self) -> None:
             self._inner_channel.close()
         ctx = ChannelContext.get_current().serialization_context
         ctx.reset_data(self._channel_id)
+
+    def num_readers(self) -> int:
+        """
+        Return the number of readers for this channel.
+        """
+        return self._inner_channel.num_readers() if self._inner_channel else 0
```

Comment:
is there an alternative we should be going to here if the inner channel doesn't exist?